testcafe/searchField.test.ts

```diff
@@ -0,0 +1,81 @@
+import { Selector } from "testcafe";
+import { Config } from "./testcafe.config";
+
+fixture`Search field test`.page`${Config.baseUrl}`
+    .beforeEach( async t => {
+        const nextButton = Selector("a").withAttribute("role", "button").child().withText("Weiter");
+        const websiteText = Selector("html").textContent;
+        const titleForIntro1 = "Willkommen bei der CovMap";
+        const titleForIntro2 = "Was ist die CovMap?";
+        const titleForIntro3 = "Was zeigt mir die CovMap an?";
+        const titleTextForPostalCode = "Für Dein regionales Risiko brauchen wir noch die Postleitzahl Deines Wohnortes";
+        
+        await t.expect(websiteText).contains(titleForIntro1).click(nextButton);
+        await t.expect(websiteText).contains(titleForIntro2).click(nextButton);
+        await t.expect(websiteText).contains(titleForIntro3).click(nextButton);
+        await t.expect(websiteText).contains(titleTextForPostalCode).pressKey("1 0 1 1 5 tab space tab tab enter");
+        await t.expect(websiteText).notContains(titleTextForPostalCode);
+    })
+    .afterEach( async t => {
+        // place holder
+    });
+
+test("find search field", async (t) => {
+    var searchField = Selector('#autocomplete').withAttribute("placeholder", "PLZ oder Landkreis")
+    await t.expect(searchField.exists).ok();
+    await t.click(searchField).typeText(searchField, "Stuttgart").pressKey("enter");
+    // wait until the map has scrolled.
+    await t.wait(1000);
+
+    // open panel for risk description
+    var riskButton = Selector('button').withAttribute("aria-label", "show more").withAttribute("aria-expanded","false")
+    await t.expect(riskButton.exists).ok();
+    const websiteText = Selector("html").textContent;
+    await t.click(riskButton).expect(websiteText).contains("Kontaktverhalten der Bevölkerung");
+
+    // close panel for risk description
+    var closeRiskButton = Selector('button').withAttribute("aria-label", "show more").withAttribute("aria-expanded","true")
+    // console.log("close button"+(await closeRiskButton.textContent) + " " + closeRiskButton.count);
+    await t.expect(closeRiskButton.exists).ok();
+    // need to access the second element for closing.
+    await t.click(closeRiskButton.nth(1)).expect(websiteText).notContains("Kontaktverhalten der Bevölkerung");;
+
+    await t.wait(1000);
+});
+
+test("Hamburger menu", async (t) => {
+
+    const websiteText = Selector("html").textContent;
+    // open menu panel
+    var openHamburgerMenu = Selector('button').withAttribute("aria-label", "Main Menu").withAttribute("aria-controls","menu-appbar")
+    var closeHamburgerMenu = Selector('button').withAttribute("aria-label", "Close Main Menu").withAttribute("aria-controls","menu-appbar")
+
+    console.log("websiteText "+ (await closeHamburgerMenu.textContent)+" "+(await closeHamburgerMenu.visible));
+
+    
+    // hanburger menu is visible
+    await t.expect(openHamburgerMenu.exists).ok()
```

Comment:
- As stated above: we do not have to check if an element exists if we interact with it later on.
- The same is true for the visibility: we can only click on visible elements. So also this check might not be necessary everywhere.
- I would only write one assertion/expectation into one `await t.` for better readability.

Reply:
- are you sure that only invisible element may be clicked? I will first try it and verify that the error message is clear enough.
- DevExpress use several `except` for `await t.` . Selector("html"). foe example: https://devexpress.github.io/testcafe/documentation/guides/basic-guides/select-page-elements.html#check-if-an-element-exists. I think it's good readable when the text is correctly formatted. not to much `await` would be my preferred way.

Reply:
With the patches above, I think with conversation could be closed.